src/main/java/ir/sahab/kafkarule/KafkaRule.java

```diff
@@ -175,21 +178,17 @@ public void createTopic(String topicName) {
     }
 
     public void createTopic(String topicName, int numPartitions) {
-        ZkClient zkClient = null;
-        ZkUtils zkUtils;
-        try {
-            // We have to pass the serializer class to ZkClient constructor Otherwise createTopic will return
-            // without error. The topic will exist in zookeeper and be returned when listing topics, but Kafka
-            // itself does not create the topic.
-            zkClient = new ZkClient(zkAddress, 30000, 30000, ZKStringSerializer$.MODULE$);
-            zkUtils = new ZkUtils(zkClient, new ZkConnection(zkAddress), JaasUtils.isZkSecurityEnabled());
-            logger.info("Executing create Topic: " + topicName + ", partitions: " + numPartitions
+        try (KafkaZkClient zkClient = KafkaZkClient
+                .apply(zkAddress, JaasUtils.isZkSaslEnabled(), 30000, 30000, MAX_IN_FLIGHT_REQUESTS,
+                        Time.SYSTEM, METRIC_GROUP, METRIC_TYPE, null, new Some<>(new ZKClientConfig()))) {
```

Comment:
```suggestion
                        Time.SYSTEM, METRIC_GROUP, METRIC_TYPE, Option.empty(), Option.empty()) {
```

When some parameter is of type Optional (or Option in Scala), you should never pass null. You should say you have no option. Many APIs just throw an exception if you pass a null value for an argument of type _Optional_. And for the second parameter (ZkClientConfig), you want to say 'I have no special configuration'. So just do not fill this parameter too.

Reply:
Done